node/consensus/src/lib.rs

```diff
@@ -431,6 +445,50 @@ impl<N: Network> Consensus<N> {
         Ok(())
     }
 
+    #[cfg(feature = "metrics")]
+    fn add_transmission_latency_metric(&self, next_block: &Block<N>) {
+        const AGE_THRESHOLD_SECONDS: i32 = 30 * 60; // 30 minutes set as stale transmission threshold
+
+        let mut keys_to_remove = Vec::new();
+        let mut transmission_queue_timestamps = self.transmissions_queue_timestamps.lock();
+
+        let solution_ids: std::collections::HashSet<_> = next_block.solutions().solution_ids().collect();
+        let transaction_ids: std::collections::HashSet<_> = next_block.transaction_ids().collect();
+
+        for (key, timestamp) in transmission_queue_timestamps.iter() {
+            let elapsed_time = std::time::Duration::from_secs((snarkos_node_bft::helpers::now() - *timestamp) as u64);
+
+            if elapsed_time.as_secs() > AGE_THRESHOLD_SECONDS as u64 {
+                // This entry is stale-- remove it from transmission queue and record it as a stale transmission
+                metrics::increment_counter(metrics::consensus::STALE_UNCONFIRMED_TRANSMISSIONS);
+                keys_to_remove.push(key.clone());
+            } else {
+                let transmission_type = match key {
+                    TransmissionID::Solution(solution_id) if solution_ids.contains(solution_id) => Some("solution"),
+                    TransmissionID::Transaction(transaction_id) if transaction_ids.contains(transaction_id) => {
+                        Some("transaction")
+                    }
+                    _ => None,
+                };
+
+                if let Some(transmission_type_string) = transmission_type {
+                    metrics::histogram_label(
+                        metrics::consensus::TRANSMISSION_LATENCY,
+                        "transmission_type",
+                        transmission_type_string.to_string(),
```

Comment:
```suggestion
                        transmission_type_string.to_owned(),
```
just to not suggest there is any conversion involved